src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenceResource.java

```diff
@@ -172,4 +173,30 @@ public ResponseEntity<Void> updateOffences(@RequestBody final  List<OffenceDto>
         service.updateOffences(offences);
         return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
     }
+
+    @PostMapping("/link-to-schedule")
+    @Operation(summary = "Link offence to schedule", description = "Requires UPDATE_OFFENCE_SCHEDULES role")
+    @ApiResponses({
+        @ApiResponse(responseCode = "201", description = "Offences linked to schedules successfully"),
+        @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.")
+    })
+    @PreAuthorize("hasRole('UPDATE_OFFENCE_SCHEDULES') and hasAuthority('SCOPE_write')")
+    public ResponseEntity<Void> linkOffencesToSchedules(@RequestBody final List<OffenceToScheduleMappingDto> offencesToSchedules) {
+        log.info("Request received to link offences to schedules");
+        service.linkOffencesToSchedules(offencesToSchedules);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
```

Comment:
you could use the response code (?) annotation and just "return" void rather than a ResponseEntity

Reply:
Agreed and done